service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.fpl.service.email.content;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.service.email.content.AbstractEmailContentProviderTest.BASE_URL;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.populatedCaseDetails;
+import static uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper.formatCaseUrl;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = { JacksonAutoConfiguration.class, C2UploadedEmailContentProvider.class })
+@TestPropertySource(properties = {"ccd.ui.base.url=" + BASE_URL})
```

Comment:
interesting... curious question should this and related tests be integration tests? 🤔 @tomaszpowroznik / @akomisarek